### DISCUSSION TITLE
React 中使用全域 sass variable 及 mixin

### DISCUSSION THREAD

**Discussion by YuChing0923** [Lidemy/forum#255] [前端相關問題]
最近在處理 React 專案，專案使用 sass 處理 style（別問為什麼不用 styled components QQ）
簡單的檔案結構如下：
```
-src
|-components
 |-Content.js
 |-Content.scss
|-sass
 |-_mixins.scss
 |-_variables.scss
|-App.js
|-App.scss
```

目前處理的方式是將 style 引入到 main.scss 再從 App 引入，如下：
App.scss
```
@import 'sass/mixins';
@import 'sass/variables';
import './Content.scss';
```
App.js
```
import './App.scss';
```
---
未來希望將結構改成 sass 分別拆分至各個不同 Component 的檔案，如下：
Content.scss
```
.content {
  color: $color-primary; // $color-primary 為 'sass/variables' 中的變數
}
```
Content.js
```
import './Content.scss';
```
App.scss
```
@import 'sass/mixins';
@import 'sass/variables';
```
App.js
```
import './App.scss';
```
目前這個作法 sass 檔在讀取到 $color-primary 的時候會沒辦法吃到變數
還是必須在每個需要用到變數的檔案中 import 'sass/variables'

想請問目前有沒有作法是能夠避免重複 import ，就能夠在個別的檔案中使用到變數及 mixin 的

---
另外關於在 React 中使用 sass 這件事，使用 React 的多數人似乎都是使用 styled components
想請問 React 中 style 的處理方式，比起一般的 css 或 sass，是不是比較建議使用 styled components
而在 mixin 中能處理顏色變化（darken, lighten）之類的功能，如果寫在 styled components 中，通常都會如何處理呢？

以上問題在麻煩大家了，謝謝各位QQ

**Comment by aszx87410** [comment]
沒在 react 中用過 sass，所以第一個問題無法回答QQ 只能回第二個

> 而在 mixin 中能處理顏色變化（darken, lighten）之類的功能，如果寫在 styled components 中，通常都會如何處理呢？

因為寫 styled component 其實就是在寫 CSS+JS，既然都有了 JS，你講的功能可能就會直接寫一隻 JS function 來處理，要用的時候 import 進來或是放在 context 裡面應該都行
